### PR TITLE
Adding end-to-end prediction

### PR DESCRIPTION
**Pull request by jiahaochen666**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

This PR mainly adding an end-to-end prediction script for users to recognize a list of images into  documents.

## Test Plan

Running the following command and check the output docx files under `./inferrence_results`
```shell
python tools/infer/text/predict_e2e.py  --image_dir ./configs/layout/yolov8/images/example.jpg --det_algorithm DB++ --rec_algorithm SVTR_PPOCRv3_CH
```

### CODE REVIEW

File: tools/infer/text/README_CN.md [hunk 1]
```diff
@@ -370,6 +366,66 @@ python tools/infer/text/predict_layout.py  --image_dir {path_to_img or dir_to_im
...
+要对输入图像或目录中的多个图像运行文档分析（即检测所有文本区域、表格区域、图像区域，并对这些区域进行文字识别，最终将结果按照图像原来的排版方式转换成Docx文件），请运行：
+
+```shell
+python tools/infer/text/predict_e2e.py --image_dir {path_to_img or dir_to_imgs} \
+                                                    --det_algorithm {DET_ALGO} \
+                                                    --rec_algorithm {REC_ALGO}
+```
+>注意：如果要可视化版面分析、表格识别和文字识别的结果，请设置`--visualize_output True`。
+
+运行后，推理结果保存在`{args.draw_img_save_dir}/system_results.txt`中，其中`--draw_img_save_dir`是保存结果的目录，这是`./inference_results`的默认设置。下面是一些结果的例子。
```
Comment (alien-0119) [inline tools/infer/text/README_CN.md:380]: 推理结果不是在system_results.txt，而是在xxx_e2e_result.txt里面

[the remark this answers is not in the document]
Comment (hongziqi) [reply tools/infer/text/README_CN.md]: 已更改

File: tools/infer/text/predict_e2e.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    if recovery:
+        final_results = sorted_layout_boxes(final_results, w_ori)
+        convert_info_docx(final_results, save_folder, f"{img_name}_converted_docx")
+
+    return final_results
+
+
+def main():
+    # from mindocr.utils.logger import set_logger
+    # set_logger(name="mindocr")
```
[the remark this answers is not in the document]
Comment (hongziqi) [reply tools/infer/text/predict_e2e.py]: 已取消注释

File: tools/infer/text/predict_e2e.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        cropped_img = image[int(top) : int(bottom), int(left) : int(right)]
+
+        if (category_id == 1 or category_id == 2 or category_id == 3) and text_system is not None:
+            start_time = time.time()
+
+            # only add white padding for text, title and list images for better recognition
+            if layout_analyzer is not None:
+                cropped_img = add_padding(cropped_img, padding_size=10, padding_color=(255, 255, 255))
+
+            rec_res_all_crops = text_system(cropped_img, do_visualize=do_visualize)
```
[the remark this answers is not in the document]
Comment (hongziqi) [reply tools/infer/text/predict_e2e.py]: 已更改

File: tools/infer/text/README_CN.md [hunk 4]
```diff
@@ -370,6 +366,66 @@ python tools/infer/text/predict_layout.py  --image_dir {path_to_img or dir_to_im
 
 算法网络在`tools/infer/text/predict_layout.py`中定义。
 
+## 端到端文档分析及恢复
+
+要对输入图像或目录中的多个图像运行文档分析（即检测所有文本区域、表格区域、图像区域，并对这些区域进行文字识别，最终将结果按照图像原来的排版方式转换成Docx文件），请运行：
+
+```shell
+python tools/infer/text/predict_e2e.py --image_dir {path_to_img or dir_to_imgs} \
```
Comment (CaitinZhao) [inline tools/infer/text/README_CN.md:374]: 名字改下，叫predict_table_e2e

Comment (hongziqi) [reply tools/infer/text/README_CN.md]: 已更改（备注：涉及到table模型的统一加table进行标注区分）

File: tools/infer/text/predict_table_e2e.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    if layout_analyzer is not None:
+        results = layout_analyzer(img_path, do_visualize=do_visualize)
+    else:
+        results = [{"category_id": 1, "bbox": [0, 0, image.shape[1], image.shape[0]], "score": 1.0}]
+
+    logger.info(f"Infering {len(results)} detected regions in {img_path}")
+
+    # crop text regions
+    h_ori, w_ori = image.shape[:2]
+    category_dict = {1: "text", 2: "title", 3: "list", 4: "table", 5: "figure"}
```
Comment (SamitHuang) [inline tools/infer/text/predict_table_e2e.py:160]: allow config?

Comment (hongziqi) [reply tools/infer/text/predict_table_e2e.py]: 已在config 增加配置layout_category_dict_path。默认配置为mindocr/utils/dict/layout_category_dict.txt(新增)

**Review by SamitHuang — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**